### PR TITLE
Added configdir parser

### PR DESCRIPTION
**Pull request by six8**
- Variable placeholder interpolation
- Travis CI configuration

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -148,11 +148,21 @@ Parsed ConfigDir
...
   }
 
 
+Python Usage
+------------
+
+.. code-block:: python
+
+  from configdir import configdir
+  config = configdir()
```
Comment (six8) [inline README.rst:162]: I decided against `import configdir; configdir()` because that was a lot of magic.

File: tests/test_configdir.py [hunk 2]
```diff
@@ -1,5 +1,150 @@
...
+    - Can be overridden with env var
+    """
+    def _just_return_dir(directory):
+        return directory
+    monkeypatch.setattr("configdir.parse", _just_return_dir)
+
+    assert configdir() == "/configdir"
+    assert configdir("/foo") == "/foo"
+
+    monkeypatch.setitem(os.environ, "CONFIGDIR", "/bar")
```
Comment (BatsShadow) [inline tests/test_configdir.py:149]: monkeypatch.setenv("CONFIGDIR", "/bar")

File: configdir/interpolator.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            values (dict): Values that placeholders will be replaced with.
+        """
+        self.values = values
+
+    def _key_segments(self, full_key):
+        """
+        Yields:
+             tuple[str, str|int, list[str]]: Key segment prefix ("." or "@"), key
+                segment value, and list of current key path segments.
+        """
```
Comment (BatsShadow) [inline configdir/interpolator.py:35]: Not sure if an example is needed or what, but this docstring does not really help me understand what this is for. Specifically, I'm not completely following the `path segments` part.

**Review by BatsShadow — APPROVED** [review]
The interpolator is a little hazy to me, but seems ok. The rest looks good.

File: tests/test_configdir.py (hunk 2 above)
Comment (six8) [reply tests/test_configdir.py:149]: That doesn't work with py2 and 3. In py2 it complains that values must be bytes. In py3 it complains values must be str. I didn't want to have a `if PY3` switch.

File: configdir/interpolator.py (hunk 3 above)
Comment (six8) [reply configdir/interpolator.py:35]: `some.key@1` has path segments `[".", "some"]`, then `[".", "some", ".", "key"]` then `[".", "some", ".", "key", "@", 1]`

**Pull request merged**